action/protocol/poll/util.go

```diff
@@ -168,16 +168,24 @@ func setCandidates(
 		return errors.New("put poll result height should be epoch start height")
 	}
 	loadCandidatesLegacy := featureCtx.LoadCandidatesLegacy(height)
+	accountCreationOpts := []state.AccountCreationOption{}
+	if protocol.MustGetFeatureCtx(ctx).CreateLegacyNonceAccount {
+		accountCreationOpts = append(accountCreationOpts, state.LegacyNonceAccountTypeOption())
+	} else {
+		accountCreationOpts = append(accountCreationOpts, state.DelegateCandidateOption())
```

Comment:
Candidate is marked on l187

Reply:
187 only applies to `if protocol.MustGetFeatureCtx(ctx).CreateLegacyNonceAccount`. The original implementation is weird: it first created an account with is not a candidate, then updated it to a candidate.
The current implementation is backward compatible.